From ef9de447a0c1a8506e891b5b6d054c1c0f829aad Mon Sep 17 00:00:00 2001
From: Yuchou Gan <yuchou.gan@nxp.com>
Date: Sat, 21 Apr 2018 01:33:15 +0800
Subject: [PATCH 3679/5242] MLK-18101-3 gpu: imx: imx8_dprc: Set has_aux_prg
 as true for _BLIT1

commit  72b35c2c4d2f6a73090e3e947fdbed39cc398434 from
https://source.codeaurora.org/external/imx/linux-imx.git

On qxp b0 board, _BLIT1 will have aux_prg if enable PRG0_SEL.

Signed-off-by: yuchou gan <yuchou.gan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/imx8_dprc.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/imx8_dprc.c b/drivers/gpu/imx/imx8_dprc.c
index 26cdef7..a440c3c 100644
--- a/drivers/gpu/imx/imx8_dprc.c
+++ b/drivers/gpu/imx/imx8_dprc.c
@@ -786,8 +786,11 @@ static int dprc_probe(struct platform_device *pdev)
 	}
 
 	switch (dprc->sc_resource) {
-	case SC_R_DC_0_BLIT0:
 	case SC_R_DC_0_BLIT1:
+		if (dprc->devtype->has_fixup)
+			dprc->has_aux_prg = true;
+		/* fall-through */
+	case SC_R_DC_0_BLIT0:
 	case SC_R_DC_1_BLIT0:
 	case SC_R_DC_1_BLIT1:
 		dprc->is_blit_chan = true;
-- 
1.7.9.5

